clients/imodels-client-management/src/base/internal/AxiosRestClient.ts

```diff
@@ -4,75 +4,91 @@
  *--------------------------------------------------------------------------------------------*/
 import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
 
-import { ContentType, HttpGetRequestParams, HttpRequestParams, HttpRequestWithBinaryBodyParams, HttpRequestWithJsonBodyParams, RestClient } from "../types/RestClient";
+import { Dictionary } from "../types";
+import { ContentType, HttpGetRequestParams, HttpRequestParams, HttpRequestWithBinaryBodyParams, HttpRequestWithJsonBodyParams, HttpResponse, HttpResponseHeaders, RestClient } from "../types/RestClient";
 
 /**
  * Function that is called if the HTTP request fails and which returns an error that will be thrown by one of the
  * methods in {@link RestClient}.
  */
 export type ParseErrorFunc = (response: { statusCode?: number, body?: unknown }, originalError: Error & { code?: string }) => Error;
 
+/** A factory for creating {@link HttpResponseHeaders} from {@link AxiosResponse}. */
+export interface HttpResponseHeadersFactory {
+  create(response: AxiosResponse): HttpResponseHeaders;
+}
+
 /** Default implementation for {@link RestClient} interface that uses `axios` library for sending the requests. */
 export class AxiosRestClient implements RestClient {
   private _parseErrorFunc: ParseErrorFunc;
+  private _responseHeadersFactory: HttpResponseHeadersFactory;
 
-  constructor(parseErrorFunc: ParseErrorFunc) {
+  constructor(parseErrorFunc: ParseErrorFunc, responseHeadersFactory: HttpResponseHeadersFactory) {
```

Comment:
Since this class is internal (not exported from the package) and also `axios` specific, we can just accept `AxiosHttpResponseHeadersFactory` here. `HttpResponseHeadersFactory` is not needed since `RestClient` interface should just tell the users what http request sending methods we need, not how they should be implemented.